FillableFieldworkAIDriver.lua

```diff
@@ -244,3 +249,92 @@ function FillableFieldworkAIDriver:setLightsMask(vehicle)
 		vehicle:setLightsTypesMask(courseplay.lights.HEADLIGHT_FULL)
 	end
 end
+
+function FillableFieldworkAIDriver:setLoadingState(object,fillUnitIndex,fillType,trigger)
+	if object and fillUnitIndex then 
+		self.fillableObject = {}
+		self.fillableObject.object = object
+		self.fillableObject.fillUnitIndex = fillUnitIndex
+		self.fillableObject.fillType = fillType
+		self.fillableObject.trigger = trigger
+	else
+		self.fillableObject = nil
+	end
+	AIDriver.setLoadingState(self)
+end
+
+function FillableFieldworkAIDriver:isFilledUntilPercantageX(currentFillType,maxFillLevel)
+	local fillLevelInfo = {}
+	self:getAllFillLevels(self.vehicle, fillLevelInfo)
+	for fillType, info in pairs(fillLevelInfo) do
+		if fillType == currentFillType then 
+			local fillLevelPercentage = info.fillLevel/info.capacity*100
+			if fillLevelPercentage >= maxFillLevel then
+				return true
+			end
+		end
+	end
+end
+
+function FillableFieldworkAIDriver:checkFilledUnitFillPercantage()
+	local fillTypeData, fillTypeDataSize= self:getSiloSelectedFillTypeData()
+	if fillTypeData == nil then
+		return
+	end
+	local fillLevelInfo = {}
+	local okFillTypes = 0
+	self:getAllFillLevels(self.vehicle, fillLevelInfo)
+	for fillType, info in pairs(fillLevelInfo) do	
+		if fillTypeData then 
+			for _,data in ipairs(fillTypeData) do
+				if data.fillType == fillType then
+					local fillLevelPercentage = info.fillLevel/info.capacity*100
+					if data.maxFillLevel and fillLevelPercentage >= data.maxFillLevel then 
+						if self.fillableObject and self.fillableObject.fillType == fillType then
+							self:forceStopLoading()
+						end
+						okFillTypes=okFillTypes+1
+					end
+				end
+			end
+		end
+	end
+	if okFillTypes == #fillTypeData then 
+		return true
+	end
+end
+
+--TODO might change this one 
+function FillableFieldworkAIDriver:levelDidNotChange(fillLevelPercent)
+	--fillLevel changed in last loop-> start timer
+	if self.prevFillLevelPct == nil or self.prevFillLevelPct ~= fillLevelPercent then
+		self.prevFillLevelPct = fillLevelPercent
+		courseplay:setCustomTimer(self.vehicle, "fillLevelChange", 3)
+	end
+	--if time is up and no fillLevel change happend, return true
+	if courseplay:timerIsThrough(self.vehicle, "fillLevelChange",false) then
+		if self.prevFillLevelPct == fillLevelPercent then
+			return true
+		end
+		courseplay:resetCustomTimer(self.vehicle, "fillLevelChange",nil)
+	end
+end
+
+function FillableFieldworkAIDriver:getSiloSelectedFillTypeSetting()
+	if self.vehicle.cp.driver:is_a(FillableFieldworkAIDriver) then
```

Comment:
See above, this should be handled by inheritance, defining getSiloSelectedFillTypeSetting also in FieldSupplyAIDriver